packages/ember-runtime/lib/system/core_object.js

```diff
@@ -430,7 +431,7 @@ CoreObject.PrototypeMixin = Mixin.create({
   */
   isDestroying: descriptor({
     get() {
-      return meta(this).isSourceDestroying();
+      return peekMeta(this).isSourceDestroying();
```

Comment:
While I totally agree that the meta is pre-initialized it does seem odd to use `peekMeta` without handling the case where there is no meta found. Is there a benefit to using peekMeta here?

Reply:
since it is guaranteed to have meta why not to use `peekMeta`,  benefit is just less work :P done